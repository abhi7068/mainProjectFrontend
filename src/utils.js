import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import { createMemoryHistory } from 'history';
import thunk from 'redux-thunk';
import PropTypes from 'prop-types';
import reducer from '../src/store/reducers/rootReducer';
import { Router } from 'react-router-dom';

function render(
  ui,
  {
    initialState,
    store = createStore(reducer, initialState, applyMiddleware(thunk)),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    const history = createMemoryHistory();
    return (
      <Provider store={store}>
        <Router history={history}>{children}</Router>
      </Provider>
    );
  }
  Wrapper.propTypes = {
    children: PropTypes.elementType,
  };
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}
export * from '@testing-library/react';
export { render };
