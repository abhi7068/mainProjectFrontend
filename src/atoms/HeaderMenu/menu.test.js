import { render } from '../../utils';
import '@testing-library/jest-dom/extend-expect';
import Menu from './index';

import React from 'react';

it('render it is', () => {
  const { findByText } = render(<Menu />);
  findByText('Profile');
});
it('render it is', () => {
  const { findByText } = render(<Menu />);
  findByText('Redeem');
});
it('render it is', () => {
  const { findByText } = render(<Menu />);
  findByText('Login | Signup');
});
it('render it is', () => {
  const { findByText } = render(<Menu />);
  findByText('Login');
});
it('displays', () => {
  const { getByTestId } = render(<Menu />);
  const anchorEl = getByTestId('anchorEl');
  expect(anchorEl.children);
});
it('displays', () => {
  const { getByTestId } = render(<Menu />);
  const isSnack = getByTestId('isSnack');
  expect(isSnack.children);
});

it('should check if message is displayed when button is clicked', () => {
  const { asFragment } = render(<Menu />);

  expect(asFragment(<Menu />)).toMatchSnapshot();
});
