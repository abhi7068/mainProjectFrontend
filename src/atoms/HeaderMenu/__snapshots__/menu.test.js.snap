// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should check if message is displayed when button is clicked 1`] = `
<DocumentFragment>
  <div
    data-testid="isSnack"
  >
    <button
      aria-controls="simple-menu"
      aria-haspopup="true"
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      id="button1"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          style="color: white;"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
